packages/flutter_tools/lib/src/template.dart

```diff
@@ -108,6 +108,16 @@ class Template {
         }
         relativeDestinationPath = relativeDestinationPath.replaceAll('$platform-$language.tmpl', platform);
       }
+
+      final bool android = context['android'] as bool;
+      if (relativeDestinationPath.contains('android') && !android) {
+        return null;
+      }
+
+      // TODO(cyanglaz): do not add iOS folder by default when 1.20.0 is released.
```

Comment:
I think we need to wait until well after 1.20 is released to give the plugin consumers time to upgrade (not just the plugin author)